### PR TITLE
feat: social login

### PR DESCRIPTION
**Pull request by gary02**

### CODE REVIEW

File: src/mutations/user/socialLogin.ts [hunk 1]
```diff
@@ -1,14 +1,56 @@
...
+    if (nonce === undefined) {
+      throw new UserInputError('nonce is required')
+    }
+    const userInfo = await userService.fetchGoogleUserInfo(
+      authorizationCode,
+      nonce
+    )
+    user = await userService.getOrCreateUserBySocialAccount({
+      socialAccountId: userInfo.id,
+      type: SOCIAL_LOGIN_TYPE.Facebook,
```
Comment (robertu7) [inline src/mutations/user/socialLogin.ts:49]: typo `SOCIAL_LOGIN_TYPE.Google`

File: src/types/user.ts [hunk 2]
```diff
@@ -993,7 +993,9 @@ export default /* GraphQL */ `
 
   input SocialLoginInput {
     type: SocialAccountType!
-    token: String!
+    authorizationCode: String!
```
Comment (robertu7) [inline src/types/user.ts:996]: write a light comment for explanation?

File: db/migrations/20230822101430_create_social_account_table.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const table = 'social_account'
+
+exports.up = async (knex) => {
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id').primary()
+    t.bigInteger('user_id').unsigned().notNullable()
+    t.enu('type', ['Google', 'Twitter', 'Facebook']).notNullable()
+    t.string('social_account_id').notNullable()
```
Comment (robertu7) [inline db/migrations/20230822101430_create_social_account_table.js:11]: or `provider_account_id`?

**Review by robertu7 — APPROVED** [review]

File: src/connectors/userService.ts [hunk 4]
```diff
@@ -2169,6 +2178,250 @@ export class UserService extends BaseService {
...
+      await this.postRegister(user)
+    }
+    return user
+  }
+
+  private getSocialAccount = async ({
+    type,
+    providerAccountId,
+  }: SocialAccount) => {
+    return await this.knex('social_account')
```
Comment (tx0c) [inline src/connectors/userService.ts:2248]: this `await` seems unnecessary;  can also omit the brace and `return`

File: src/connectors/userService.ts [hunk 5]
```diff
@@ -2169,6 +2178,250 @@ export class UserService extends BaseService {
...
+   */
+  public fetchTwitterUserInfo = async (
+    authorizationCode: string,
+    codeVerifier: string
+  ) => {
+    const accessToken = await this.exchangeTwitterAccessToken(
+      authorizationCode,
+      codeVerifier
+    )
+    return await this.fetchTwitterUserInfoByAccessToken(accessToken)
```
Comment (tx0c) [inline src/connectors/userService.ts:2281]: `await` in the last return is always unnecessary

Comment (gary02) [reply src/connectors/userService.ts:2281]: weird, npm run format && npm run lint:fix should fix this

**Review by robertu7 — APPROVED** [review]

**Pull request merged**